plugins/catalog/alpha-api-report.md

```diff
@@ -3,17 +3,72 @@
 > Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
 
 ```ts
+/// <reference types="react" />
+
+import { AnyExtensionInputMap } from '@backstage/frontend-plugin-api';
 import { BackstagePlugin } from '@backstage/frontend-plugin-api';
 import { Extension } from '@backstage/frontend-plugin-api';
+import { PortableSchema } from '@backstage/frontend-plugin-api';
 
 // @alpha (undocumented)
 export const CatalogApi: Extension<{}>;
 
+// @alpha (undocumented)
+export const CatalogEntityKindFilter: Extension<{
+  initialFilter: string;
+}>;
+
+// @alpha (undocumented)
+export const CatalogEntityLifecycleFilter: Extension<never>;
+
+// @alpha (undocumented)
+export const CatalogEntityNamespaceFilter: Extension<never>;
+
+// @alpha (undocumented)
+export const CatalogEntityOwnerFilter: Extension<{
+  mode?: 'all' | 'owners-only' | undefined;
+}>;
+
+// @alpha (undocumented)
+export const CatalogEntityPage: Extension<{
+  path: string;
+}>;
+
+// @alpha (undocumented)
+export const CatalogEntityProcessingStatusFilter: Extension<never>;
+
+// @alpha (undocumented)
+export const CatalogEntityTagFilter: Extension<never>;
+
+// @alpha (undocumented)
+export const CatalogEntityTypeFilter: Extension<never>;
+
+// @alpha (undocumented)
+export const CatalogIndexPage: Extension<{
+  path: string;
+}>;
+
 // @alpha (undocumented)
 export const CatalogSearchResultListItemExtension: Extension<{
   noTrack?: boolean | undefined;
 }>;
 
+// @alpha (undocumented)
+export const CatalogUserListFilter: Extension<{
```

Comment:
General comment not directly related to this PR: I think this export pattern would be very confusing in the long run, plenty of overlap with React components. I think the only way around it long term is if we started to move all component exports to something like `/components`.

What I'm leaning towards for now is to not export any extensions at all, it's not like you can do much with them at this point anyway 😅. Main reason for now would probably be testing, but I would find it a bit unfortunate if packages where cherry-picking extensions from a different plugin, and besides, we could just disable extensions by default from a plugins and you can pick the ones you want that way. By removing the exports we can start seeing if there is actually a need to reference extensions directly at all.

Reply:
That's a good call out! At first, I thought that having access to the `at` property would help in overriding the extensions to avoid using magic strings for the attachment point, but after your comment, I realized that we already have access to the extensions through the plugin instance, like:
```tsx
import catalogPlugin from '@backstage/plugin-catalog';
const fistExtension = catalogPlugin.extensions[0].id;
```
I'll remove the exports for now 🙌🏻

Reply:
Here os the commit 🙂 : https://github.com/backstage/backstage/pull/20355/commits/579279f1746b12b6e0bd34db49b3cc46cd15d1e0